src/helm/benchmark/annotation/bigcodebench_annotator.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any, List
+
+from helm.benchmark.adaptation.request_state import RequestState
+from helm.benchmark.annotation.annotator import Annotator
+from helm.common.request import Request
+from gradio_client import Client, handle_file
+from tempfile import TemporaryDirectory
+
+from helm.common.hierarchical_logger import hlog
+
+import ast
+import traceback
+import time
+import json
+
+
+def syntax_check(code, verbose=False):
+    try:
+        ast.parse(code)
+        return True
+    except (SyntaxError, MemoryError):
+        if verbose:
+            traceback.print_exc()
+        return False
+
+
+def code_extract(text: str) -> str:
+    lines = text.split("\n")
+    longest_line_pair = (0, 0)
+    longest_so_far = 0
+
+    for i in range(len(lines)):
+        for j in range(i + 1, len(lines)):
+            current_lines = "\n".join(lines[i : j + 1])
+            if syntax_check(current_lines):
+                current_length = sum(1 for line in lines[i : j + 1] if line.strip())
+                if current_length > longest_so_far:
+                    longest_so_far = current_length
+                    longest_line_pair = (i, j)
+
+    return "\n".join(lines[longest_line_pair[0] : longest_line_pair[1] + 1])
+
+
+class BigCodeBenchAnnotator(Annotator):
+    """The BigCodeBench autograder."""
+
+    name = "bigcodebench"
+
+    def __init__(self):
+        self.remote_execute_api = "https://bigcode-bigcodebench-evaluator.hf.space/"
+        self.split = "instruct"
+        self.subset = "full"
+        self.pass_k = "1"  # Original: "1,5,10"
+        self.is_macro = True
+
+    def annotate(self, request_state: RequestState) -> Any:
+        pass
+
+    def annotate_all(self, request_states: List[RequestState]) -> Any:
+        assert all(request_state.result for request_state in request_states)
+        assert all(len(request_state.result.completions) == 1 for request_state in request_states)
+        assert all(request_state.instance.extra_data for request_state in request_states)
+
+        with TemporaryDirectory() as tmpdir:
+            # with open(f"{tmpdir}/result.jsonl", "w") as file:
+            with open(f"tmp_result.jsonl", "w") as file:
```

Comment:
Any reason you need to write to disk instead of just keeping a buffer in memory - does Gradio need this, or is the request too big to keep in memory?

Reply:
It's easier to use a file with Gradio, I tried to avoid the writing but it seems much less straightforward.